### ISSUE TITLE
DataGridNumericUpDownColumn foreground color

### ISSUE THREAD

**Issue by GAMP** [MahApps/MahApps.Metro#2434]
![new bitmap image](https://cloud.githubusercontent.com/assets/4171664/13823929/64f11e5e-ebb4-11e5-8afe-1d5dd1f7500a.jpg)

When selected font color is not changed as on other text elements.

**Comment by xxMUROxx** [comment]
Which version of *MahApps* do you use.
Can you show us the XAML code that reproduces the problem.

**Comment by GAMP** [comment]
`          <ma:DataGridNumericUpDownColumn
                            Header="{lex:Loc Key=QTY}"
                            Minimum="1"
                            Binding="{Binding Quantity,UpdateSourceTrigger=PropertyChanged,StringFormat=N2}"
                            HeaderStyle="{StaticResource _right_aligned_header}">
                            <ma:DataGridNumericUpDownColumn.CellStyle>
                                <Style TargetType="DataGridCell" BasedOn="{StaticResource {x:Type DataGridCell}}">
                                    <Setter Property="ma:NumericUpDown.IsReadOnly" Value="{Binding IsBundleProduct}"/>
                                </Style>
                            </ma:DataGridNumericUpDownColumn.CellStyle>
                        </ma:DataGridNumericUpDownColumn>`

Using latest source.

**Comment by xxMUROxx** [comment]
How is **_right_aligned_header** defined?

**Comment by GAMP** [comment]
`     <DataGrid.Resources>
                        <Style x:Key="_right_aligned_header" TargetType="DataGridColumnHeader" BasedOn="{StaticResource MetroDataGridColumnHeader}">
                            <Setter Property="HorizontalContentAlignment" Value="Right"/>
                        </Style>
                        <Style x:Key="_left_aligned_header" TargetType="DataGridColumnHeader" BasedOn="{StaticResource MetroDataGridColumnHeader}">
                            <Setter Property="HorizontalContentAlignment" Value="Left"/>
                        </Style>
                    </DataGrid.Resources>`

**Comment by xxMUROxx** [comment]
Ok, the error is in following code:

~~~xaml
<Style TargetType="DataGridCell" BasedOn="{StaticResource {x:Type DataGridCell}}">
  <Setter Property="ma:NumericUpDown.IsReadOnly" Value="{Binding IsBundleProduct}"/>
</Style>
~~~

Just change `{StaticResource {x:Type DataGridCell}}` with `{StaticResource MetroDataGridCell}`

**Comment by punker76** [comment]
@GAMP Is this still an issue for you?

**Comment by GAMP** [comment]
>     <ma:DataGridNumericUpDownColumn.CellStyle>
                                <Style BasedOn="{StaticResource MetroDataGridCell}" TargetType="DataGridCell">
                                    <Setter Property="ma:NumericUpDown.IsReadOnly" Value="{Binding IsBundleProduct}" />
                                </Style>
                            </ma:DataGridNumericUpDownColumn.CellStyle>
Did not resolve the problem. Foreground still remains black when row is selected.

**Comment by thirdwaffle** [comment]
It seems that the default style of the `NumericUpDown` control is taking precedence when the row is selected. 

Here's what [Snoop](https://snoopwpf.codeplex.com/) shows for a cell in a normal `DataGridTextColumn` when the row is selected:
![image](https://cloud.githubusercontent.com/assets/10321525/15446776/373a5172-1f6e-11e6-9631-10c69ed7596e.png)

And here's what Snoop shows for a cell in a `DataGridNumericUpDownColumn` when the row is selected:
![image](https://cloud.githubusercontent.com/assets/10321525/15446782/5f8e2c2a-1f6e-11e6-9224-899e7b44b2e7.png)

I'll keep poking around and see what if I can find a way to get the `Foreground` value to be inherited.

**Comment by GAMP** [comment]
Still an issue

**Comment by punker76** [comment]
@GAMP It's never too late to fix Issues :-D This is now fixed and available in 1.5.0
